plugins/Denizen/scripts/fun/forbidden_words.dsc

```diff
@@ -0,0 +1,72 @@
+get_forbidden_word:
+    type: procedure
+    debug: false
+    definitions: item|should_exist
+    script:
+    - if <[should_exist]> and not <server.has_flag[forbidden_words]>:
+        - determine null
+    - if <[item].material.name> != diamond or !<[item].has_display>:
+        - determine null
+    - define word <[item].display.strip_color.to_lowercase.before[ ]>
+    - if <server.flag[forbidden_words].contains[<[word]>].if_null[false]> == <[should_exist]>:
+        - determine <[word]>
+    - determine null
+
+forbidden_words:
+    type: world
+    debug: false
+    events:
+        on entity damaged by lava type:dropped_item:
+        - ratelimit <context.entity> 3t
+        - define item <context.entity.item>
+        - define word <[item].proc[get_forbidden_word].context[false]>
+        - if <[word]> == null:
+            - stop
+        - flag server forbidden_words:->:<[word]>
+        - customevent id:forbidden_word_created context:[word=<[word]>]
+        after player empties water_bucket:
+        - define diamonds <context.relative.find_entities[dropped_item].within[1.5]>
+        - if <[diamonds].is_empty>:
+            - stop
+        - define diamond <[diamonds].first>
+        - define word <[diamond].item.proc[get_forbidden_word].context[true]>
+        - if <[word]> == null:
+            - stop
+        - flag server forbidden_words:<-:<[word]>
+        - modifyblock <context.relative> air
```

Comment:
What happens if they waterlog a block? does this event trigger? if so, does this line delete whatever they waterlog?